cpp/src/arrow/io/test-common.h

```diff
@@ -47,7 +47,8 @@ class ARROW_EXPORT MemoryMapFixture {
   void CreateFile(const std::string& path, int64_t size);
 
   Status InitMemoryMap(int64_t size, const std::string& path,
-                       std::shared_ptr<MemoryMappedFile>* mmap);
+                       std::shared_ptr<MemoryMappedFile>* mmap,
+		       const off_t offset = 0);
```

Comment:
Code style quibble: adding an optional parameter after the output parameter, which should be last. Could you add this as an overload (for example `Status InitMemoryMap(int64_t offset, int64_t size, const std::string& path, std::shared_ptr<MemoryMappedFile>* mmap)`)?

Reply:
Agree, adding a overload seems more clean. Will add this in the next push. Thanks!